### PR TITLE
Make KeyVaultKeyStore can work without keyVaultUri.

### PR DESCRIPTION
**Pull request by chenrujun**
Make KeyVaultKeyStore can work without keyVaultUri.

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @lzc-1997-abel , please help to review this PR. 🙏

**Comment by chenrujun** [comment]
Hi, @yiliuTo , please help to review this PR. 🙏

File: sdk/keyvault/azure-security-keyvault-jca/src/test/java/com/azure/security/keyvault/jca/KeyVaultCertificatesTest.java [hunk 1]
```diff
@@ -51,9 +51,10 @@ public void testGetCertificate() {
     }
 
     @Test
-    public void testRefreshAndGetAliasByCertificate() {
+    public void testRefreshAndGetAliasByCertificate() throws InterruptedException {
         Assertions.assertEquals(keyVaultCertificates.refreshAndGetAliasByCertificate(certificate), "myalias");
         when(keyVaultClient.getAliases()).thenReturn(null);
+        Thread.sleep(10); // TODO: (rujche) update this logic
```
Comment (chenrujun) [inline sdk/keyvault/azure-security-keyvault-jca/src/test/java/com/azure/security/keyvault/jca/KeyVaultCertificatesTest.java:57]: Hi, @lzc-1997-abel , could you please investigate this issue?

**Comment by chenrujun** [comment]
Hi, @lzc-1997-abel , please comment `LGTM` if you agree to merge this PR.

**Comment by lzc-1997-abel** [comment]
If we support KeyVaultKeyStore work without keyVaultUri, should we change the judge conditions [here](https://github.com/Azure/azure-sdk-for-java/blob/d0450358876fdf9d7a720f6f1dc0c45f9d526563/sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java#L30)

**Comment by chenrujun** [comment]
@lzc-1997-abel , thank you very much for your careful check.  

That is in the starter module, it's no problem that starter module do nothing if keyvault uri is not set.

**Comment by lzc-1997-abel** [comment]
LGTM

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultClient.java [hunk 2]
```diff
@@ -57,49 +57,38 @@
...
      */
-    private String clientSecret;
+    private final String clientSecret;
 
     /**
-     * Stores the managed identity (either the user-assigned managed identity
-     * object ID or null if system-assigned)
+     * Stores the managed identity (either the user-assigned managed identity object ID or null if system-assigned)
      */
     private String managedIdentity;
```
Comment (yiliuTo) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultClient.java:90]: Need managedIdentity to be final?

**Review by yiliuTo — APPROVED** [review]
LGTM

**Pull request merged**